public/app/features/variables/datasource/DataSourceVariableEditor.tsx

```diff
@@ -93,49 +87,29 @@ export class DataSourceVariableEditorUnConnected extends PureComponent<Props> {
   };
 
   render() {
-    const { variable, extended, changeVariableMultiValue } = this.props;
+    const { variable, extended } = this.props;
 
     const typeOptions = extended?.dataSourceTypes?.length
       ? extended.dataSourceTypes?.map((ds) => ({ value: ds.value ?? '', label: ds.text }))
       : [];
 
-    const typeValue = typeOptions.find((o) => o.value === variable.query) ?? typeOptions[0];
-
     return (
-      <>
-        <VariableLegend>Data source options</VariableLegend>
-        <VariableSelectField
-          name="Type"
-          value={typeValue}
-          options={typeOptions}
-          onChange={this.onDataSourceTypeChanged}
-          testId={selectors.pages.Dashboard.Settings.Variables.Edit.DatasourceVariable.datasourceSelect}
-        />
-
-        <VariableTextField
-          value={this.props.variable.regex}
-          name="Instance name filter"
-          placeholder="/.*-(.*)-.*/"
-          onChange={this.onRegExChange}
-          onBlur={this.onRegExBlur}
-          description={
-            <div>
-              Regex filter for which data source instances to choose from in the variable value list. Leave empty for
-              all.
-              <br />
-              <br />
-              Example: <code>/^prod/</code>
-            </div>
-          }
-        />
-
-        <VariableLegend>Selection options</VariableLegend>
-        <SelectionOptionsEditor
-          variable={variable}
-          onPropChange={this.onSelectionOptionsChange}
-          onMultiChanged={changeVariableMultiValue}
-        />
-      </>
+      <DataSourceVariableForm
+        query={variable.query}
+        regex={variable.regex}
+        multi={variable.multi}
+        includeAll={variable.includeAll}
+        optionTypes={typeOptions}
+        onChange={this.onDataSourceTypeChanged}
+        onRegExBlur={this.onRegExBlur}
+        onMultiChange={this.onMultiChanged}
+        onIncludeAllChange={(event) =>
+          this.onSelectionOptionsChange({ propName: 'includeAll', propValue: event.currentTarget.checked })
+        }
+        onAllValueChange={(event) =>
+          this.onSelectionOptionsChange({ propName: 'allValue', propValue: event.currentTarget.value })
+        }
```

Comment:
Small nit: Probably we can move these two handlers to their own instance methods like `onMultiChanged`. Just for consistency. WDYT?